apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/test/store.test.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import waitForExpect from 'wait-for-expect';
+
+/**
+ * Internal dependencies
+ */
+import { dispatch, select } from '@wordpress/data';
+import { register } from '../store';
+
+const STORE_KEY = 'automattic/wpcom-welcome-guide';
+
+beforeAll( () => {
+	register();
+} );
+
+let originalFetch;
+beforeEach( () => {
+	dispatch( STORE_KEY ).resetStore();
+	originalFetch = window.fetch;
+	window.fetch = jest.fn();
+} );
+
+afterEach( () => {
+	window.fetch = originalFetch;
+} );
+
+test( "by default the store isn't loaded", () => {
+	const isLoaded = select( STORE_KEY ).isWelcomeGuideStatusLoaded();
+	expect( isLoaded ).toBe( false );
+} );
+
+test( 'after fetching the guide status the store is loaded', async () => {
+	window.fetch.mockResolvedValue( {
+		status: 200,
+		json: () => Promise.resolve( { show_welcome_guide: true, variant: 'tour' } ),
+	} );
+
+	dispatch( STORE_KEY ).fetchWelcomeGuideStatus();
+
+	await waitForExpect( () => {
+		const isLoaded = select( STORE_KEY ).isWelcomeGuideStatusLoaded();
+		expect( isLoaded ).toBe( true );
+	} );
+
+	expect( window.fetch ).toHaveBeenCalledWith(
+		'/wpcom/v2/block-editor/nux?_locale=user',
+		expect.anything()
+	);
+
+	// Check the store is loaded with the state that came from the server
+	const isWelcomeGuideShown = select( STORE_KEY ).isWelcomeGuideShown();
+	expect( isWelcomeGuideShown ).toBe( true );
+	const welcomeGuideVariant = select( STORE_KEY ).getWelcomeGuideVariant();
+	expect( welcomeGuideVariant ).toBe( 'tour' );
```

Comment:
Is it worth setting an initial variant of `test` and then testing that reset clears it?

Reply:
The reset action is only needed for testing though. What I which is that the `@wordpress/data` had an `unregister()` function so I could just reregister the store to take it back to defaults before each test. I probably need a comment for the reset action.

But yeah since I've had introduced code for resetting it should really be under test. I think I can probably create a generic test that just checks that all store values are taken back to their defaults.

Reply:
I like the new layout more 👍 

We could make it future-proof by adding an optional store-key argument to the `register()` function and comparing the values between a newly-created store and the reset one, but that seems way over the top unless those default values are actually going to change.

Sometimes I leave a comment like in cases like this though, along the lines of "If this test ever breaks because the defaults change, we should add the store key as an argument to `register()` so we can create a fresh test store to compare values against". ;)

Reply:
> We could make it future-proof by adding an optional store-key argument to the `register()` function

😮 didn't even think of that! I could just register a unique store for every single test and not have the reset action! That could actually work pretty well!
It does mean we'd probably be leaking memory in watch mode though ...

I should probably just go ahead and open a PR to add an `unregister()` function. I've run into this enough times that I think it's a valid use case.